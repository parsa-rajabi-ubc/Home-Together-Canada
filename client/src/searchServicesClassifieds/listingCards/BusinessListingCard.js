/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.19
 *
 * @Description: Business Listing Card
 *
 */

import React from 'react';
import PropTypes from "prop-types";
import Moment from 'react-moment';


function BusinessListingCard(props) {
    const {logo, title, businessName, shortDescription, datePosted} = props;

    return (
        <section className={"card-container mt-0"}>
            <img className={"float-left w-24 h-24 mx-4 "} src={logo}/>
            <div className={"inline align-middle "}>
                <label className={"font-semibold justify-between"}> {title} </label>
                {/*{shortDescription}*/}
                <section className={"inline float-right pr-6 justify-end"}>
                    <Moment format="MMM D, YYYY">{datePosted}</Moment>
                </section>
            </div>

            <div className={"mx-6 inline items-center justify-between leading-tight font-light"}>
                <section className={"mb-3"}> {businessName} </section>
                <section> {shortDescription} </section>

            </div>
        </section>
    );
}

BusinessListingCard.propTypes = {
    logo: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    businessName: PropTypes.string.isRequired,
    shortDescription: PropTypes.string.isRequired,
    datePosted: PropTypes.string.isRequired,
}


export default BusinessListingCard;
