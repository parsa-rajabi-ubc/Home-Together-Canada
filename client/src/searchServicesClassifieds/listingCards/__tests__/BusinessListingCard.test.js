/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.18
 *
 * @Description: Business Listing Card Test
 *
 */

import React from 'react';
import renderer from  'react-test-renderer'
import BusinessListingCard from "../BusinessListingCard";
import PropTypes from "prop-types";

describe('BusinessListingCard', () => {
    describe('Container test', () => {
        it('should match snapshot test', () => {
            //given
            const props = {
                logo: "Potato",
                title: "Soup",
                businessName: "Pasta",
                shortDescription: "Snow",
                datePosted: "2021-02-14T05:42:39.000Z",
            }
            
            //when
            const component = renderer.create(<BusinessListingCard {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});