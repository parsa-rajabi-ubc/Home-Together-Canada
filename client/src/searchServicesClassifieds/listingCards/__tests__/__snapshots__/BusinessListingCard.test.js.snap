// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessListingCard Container test should match snapshot test 1`] = `
<section
  className="card-container mt-0"
>
  <img
    alt=""
    className="float-left w-24 h-auto mx-4 "
    src="HTC_Logo.jpg"
  />
  <div
    className="inline align-middle "
  >
    <label
      className="font-semibold justify-between"
    >
       
      Soup
       
    </label>
    <section
      className="inline float-right pr-6 justify-end"
    >
      <time
        dateTime={"2021-02-14T05:42:39.000Z"}
      >
        Feb 13, 2021
      </time>
    </section>
  </div>
  <div
    className="mx-6 inline items-center justify-between leading-tight font-light"
  >
    <section
      className="mb-3"
    >
       
      Pasta
       
    </section>
    <section>
       
      Snow
       
    </section>
  </div>
</section>
`;
